wrapper/CSharp/wolfSSL-TLS-Server/wolfSSL-TLS-Server.cs

```diff
@@ -208,12 +209,22 @@ public static void Main(string[] args)
             return;
         }
 
+        /* get and print sni used by the client */
+        if (haveSNI(args)) {
+            IntPtr data = IntPtr.Zero;
+
+            ushort size = wolfssl.SNI_GetRequest(ssl, 0, ref data);
+            string dataStr = Marshal.PtrToStringAnsi(data);
+            Console.WriteLine("(SNI_GetRequest) Size of SNI used by client: " + size);
+            Console.WriteLine("(SNI_GetRequest) SNI used by client: " + dataStr);
+        }
+
         /* print out results of TLS/SSL accept */
         Console.WriteLine("SSL version is " + wolfssl.get_version(ssl));
         Console.WriteLine("SSL cipher suite is " + wolfssl.get_current_cipher(ssl));
 
         /* read and print out the message then reply */
-        if (wolfssl.read(ssl, buff, 1023) < 0)
+        if (wolfssl.read(ssl, buff, 1024) < 0)

```

Comment:
The point of reading 1023 is to leave a null terminator. Buff is 1024, so this is 1024-1. We use this same pattern in a few of the examples.